File: 1786-count-the-number-of-consistent-strings/solution.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        Set<Character> map = new HashSet<>();
        for (char i : allowed.toCharArray()) {
            map.add(i);
        }

        int result = words.length;
        for (String word : words) {
            for (char a : word.toCharArray()) {
                if (!map.contains(a)) {
                    result--;
                    break;
                }
            }
        }

        return result;
    }
}

File: 1786-count-the-number-of-consistent-strings/solution.go
func countConsistentStrings(allowed string, words []string) int {
    theMap := make([]int, 26)
    for _, v := range allowed {
        theMap[rune(v) - 'a']++
    }

    counter := 0
    for _, word := range words {
        isConsistent := true
        for _, c := range word {
            if theMap[rune(c) - 'a'] <= 0 {
                isConsistent = false
                break
            }
        }

        if isConsistent {
            counter++
        }
    }

    return counter
}
